src/java.base/share/classes/java/lang/ProcessBuilder.java

```diff
@@ -1296,6 +1296,10 @@ public static List<Process> startPipeline(List<ProcessBuilder> builders) throws
                     redirects[1] = new RedirectPipeImpl();  // placeholder for new output
                 }
                 processes.add(builder.start(redirects));
+                if (prevOutput instanceof RedirectPipeImpl redir) {
+                    // Wrap the fd so it can be closed
+                    new Process.PipeInputStream(redir.getFd()).close();
```

Comment:
It looks unusual to need to create a new input stream to invoke close but I think this is okay here.